x-pack/plugins/observability_solution/slo/public/embeddable/slo/overview/slo_configuration.scss

```diff
@@ -0,0 +1,3 @@
+.euiOverlayMask {
+  z-index: unset !important
```

Comment:
This fixes the issue with the kqlbar suggestions shown below the overlay mask of the modal. 

<img width="773" alt="320770685-5b0e9d7c-b69c-45e4-92ac-ed33ebde7519" src="https://github.com/elastic/kibana/assets/2852703/34e51b65-c8dc-43d5-8e8a-da1eb0661580">

Similar to this [one](https://github.com/elastic/eui/issues/5757). Suggested [solution](https://github.com/elastic/eui/issues/5757#issuecomment-1088265561)

Reply:
@andreadelrio Thanks for looking into this one. You are right, I had it in my todo list to fix the issue with the custom filter. I just pushed the change. Let me look at the button group issue as well

Reply:
I fixed the `button-group` padding issue as well. Let me know what you think!

<img width="400" alt="Screenshot 2024-04-11 at 21 59 10" src="https://github.com/elastic/kibana/assets/2852703/e3b86a80-bdbd-4efb-a3e0-f132839f3e46">

Reply:
@mgiota Thanks for fixing that. This override is generating some weird consequences such as:

a) The top navigation is coming to the front of the overlay when it shouldn't
<img width="1077" alt="image" src="https://github.com/elastic/kibana/assets/4016496/81a013c4-2600-4c5e-a5c5-d16064e1c373">
<img width="1440" alt="image" src="https://github.com/elastic/kibana/assets/4016496/ea03da1f-9028-4100-a8e1-657e5b2421a6">

b) In serverless the flyout's header is covered by the navigation.
<img width="1442" alt="image" src="https://github.com/elastic/kibana/assets/4016496/0537d65f-7fa8-402f-8e51-695cf880c1cf">

You might want to consult with @elastic/eui-team if there's a better way to achieve what you are trying to achieve.

Reply:
@andreadelrio Thanks a lot for looking into it thoroughly! I fixed the issue for both cases you described:

<img width="400" alt="Screenshot 2024-04-12 at 11 25 09" src="https://github.com/elastic/kibana/assets/2852703/00724c88-5562-498b-85f6-6b9dd190016a">


<img width="400" alt="Screenshot 2024-04-12 at 11 23 10" src="https://github.com/elastic/kibana/assets/2852703/6b0ab221-8170-4986-9620-e6bcc93a4083">

I definitely plan to reach out to @elastic/eui-team and possibly unified Search team, since I have more questions for them. In a nutshell, I had 2 options for the initial problem with the kql bar suggestions showing below the overlay mask:
- either increase the z-index of the `Searchbar`. Unfortunately using `css` directly in the unified `SearchBar` component is not possible. The whole page was breaking, telling me that SearchBar is not defined. So I left this solution, which I would ideally prefer to use.
- or decrease the z-index of the overlayMask to make suggestions bar visible. This is the approach I followed, now it works with the z-index I provide


Hopefully I can get your approval. It is a cool feature we would like to end up in the following release. Thanks for your support!

Reply:
⚠️ Please don't override the CSS of EUI global components without adding some sort of modifier that clearly indicates you only want to apply it to **this specific usage** - otherwise the change cascades down everywhere (and I believe to multiple plugins??) and causes a ton of side effects.

[Modals are meant to overlay on top of **everything**](https://eui.elastic.co/#/layout/modal/guidelines) including the global nav. They're essentially interrupt-driven dialogs. If you don't want this behavior, you should not be using an EuiModal - please consider using an [EuiFlyout](https://eui.elastic.co/#/layout/flyout) instead, which automatically sits below the global nav bar and leaves it accessible to keyboard **and** mouse users.

Reply:
++ to Cee's comments. @mgiota I believe a flyout would be better suited here. This is the component of choice for most of our panels (with the exception of those that have a dedicated editor). You can also refer to this internal [document](https://docs.google.com/document/d/1vNcHT3YgX0bZNGLILS1LtlP6nW2z806ryU2ESgYRpAM/edit?usp=sharing) where we make this recommendation `Best practices for teams adding new panels to Dashboards`

>Creation: ideally users should be able to create the new panel by displaying a flyout in the same dashboard. If this is not possible, they should find at least a link to creation (see ML example below)`. cc @elastic/kibana-presentation for awareness.

Reply:
cc @jasonrhodes This is the issue I mentioned to you with the z-index and the modal. I couldn't find a way to fix the issue of the kqlbar fixing positioned below the modal mask, without overriding the CSS of the global component. 

@andreadelrio and @cee-chen We were inspired by Machine learning embeddable regarding the use of Modal for our embeddable's configuration. But our use case is a bit more complex, since now we want to include a kql bar which causes the issue.

<img width="400" alt="Screenshot 2024-04-15 at 08 37 54" src="https://github.com/elastic/kibana/assets/2852703/1235fc08-c040-431f-945d-06930ae0cc5b">


As you described above the overlay is positioned on top of everything. I was trying to find a way to apply my change to a specific usage, but I couldn't. More specifically this is the solution I was hoping would work, but unfortunately this breaks the whole embeddable configuration. Modal doesn't even open. That's why I went with the solution of overriding the global css (making sure I don't break anything in the Dashboard app with manual testing on my side). I was expecting design team to comment on this and recommend me a better way to do this. Changing to Flyout sounds good, but it sounds like a bigger change that I was hoping for.


<img width="400" alt="Screenshot 2024-04-15 at 08 52 53" src="https://github.com/elastic/kibana/assets/2852703/2f728db9-3960-461b-ac4d-9f2d0980b055">


@jasonrhodes What options do we have here?
- remove the use of kqlbar so that we ship the grouping functionality to our users?
- replace modal with flyout?
- try to find a way to apply the change I want to the kqlbar?

The ideal solution is the 3rd one, I just couldn't find a way to do it. From my understanding this would involve unified search team and the option to support css on their component. @andreadelrio and @cee-chen Do you have something else to recommend on how to fix the z-Index issue?

Reply:
@andreadelrio @cee-chen I pushed the final fix, where I replace `Modal` with `Flyout` according to your suggestions. 

https://github.com/elastic/kibana/assets/2852703/c2975bb8-616a-4e1c-9a3f-1b486e1f232c

Let me know if it looks good and if we can get final approval for this PR.

Reply:
It looks terrific to me! I have a minor spacing suggestion for the save/cancel button locations that I'll make in a separate code commit (the below is from our docs and is likely what most flyouts use across Kibana).

<img width="756" alt="" src="https://github.com/elastic/kibana/assets/549407/4a20aa62-b4bd-4600-ad81-56192dc556bf">